newIDE/app/src/MainFrame/index.js

```diff
@@ -459,6 +459,26 @@ class MainFrame extends React.Component<Props, State> {
       .then(() => this.setState({ loadingProject: false }));
   };
 
+  closeApp = (): void => {
+    Window.quit();
+  };
+
+  askToCloseApp = (): Promise<void> | void => {
+    console.log(this.state.currentProject);
+    if (!this.state.currentProject) return this.closeApp();
+
+    const { i18n } = this.props;
+    // eslint-disable-next-line
+    const answer = confirm(
+      i18n._(
+        t`Exit GDevelop? Any changes that have not been saved will be lost.`
+      )
+    );
+    if (!answer) return Promise.resolve();
```

Comment:
Do we need this? The documentation of Electron's [app.quit states](https://www.electronjs.org/docs/all#appquit) that calling app.quit() will "Try to close all windows" and "This method guarantees that all beforeunload and unload event handlers are correctly executed".

Which means that it will correctly trigger the dialog in https://github.com/4ian/GDevelop/blob/master/newIDE/app/src/UI/CloseConfirmDialog.js no?

You may have been confused by the fact that this dialog is *not* shown in development because of this line: https://github.com/4ian/GDevelop/blob/master/newIDE/app/src/UI/CloseConfirmDialog.js#L24 :)

(otherwise, we would get the message at each reload after changing a source file)

Reply:
Oh, right! Okay, I'll fix this right now.